src/genetic_testing/routers/ncbi.py

```diff
@@ -5,36 +5,127 @@
 
-    This function searches the provided NCBI database and returns the list of UIDs matching the
-    Entrez query.
+    This function searches the provided Entrez database and returns the list of UIDs matching the Entrez query.
 
-    Args:
-        database (str): Name of the Entrez database.
-        term (str): Entrez text query.
+    Parameters
+    ----------
+    database : str
+        Name of the Entrez database.
+    term : str
+        Entrez text query.
 
-    Returns:
-        List[int]: A list of UIDs.
+    Returns
+    -------
+    ListElement[StringElement]
+       A list of UIDs.
     """
-    handle = Entrez.esearch(db=database, term=term)
+    handle = Entrez.esearch(db=database, term=term, retmax=RETMAX)
     uids = []
     if handle:
         record = Entrez.read(handle)
         uids = record["IdList"]
-
+        handle.close()
     return uids
+
+
+def _summary(
+    database: str, ids: ListElement[StringElement]
+) -> ListElement[DictionaryElement]:
+    """Get the document summaries for the input UIDs.
+
+    This function returns the document summaries for the input list of UIDs and Entrez database.
+
+    Parameters
+    ----------
+    database : str
+        Name of the Entrez database.
+    ids : ListElement[StringElement]
+        A list of UIDs.
+
+    Returns
+    -------
+    ListElement[DictionaryElement]
+        A list of dictionaries where each dictionary element is a single summary
+    """
+    handle = Entrez.esummary(db=database, id=ids, retmax=10000)
+    if handle:
+        record = Entrez.read(handle)
+        handle.close()
+    return record
+
+
+def _parse_summary(
+    document_summaries: ListElement[DictionaryElement],
+) -> List[Dict[str, str]]:
+    """Parse document summaries.
+
+    This function parses the input document summaries and selects a subset of columns for the final summary
+
+    Parameters
+    ----------
+    document_summaries : ListElement[DictionaryElement]
+       A list of dictionary elements representing document summaries.
+
+    Returns
+    -------
+    List[Dict[str, str]]
+        A list of dictionaries containing parsed document summaries.
+    """
+    parsed_summaries = []
+    for document_summary in document_summaries:
+        parsed_summaries.append(
+            {key: document_summary.get(key, None) for key in NCBI_SUMMARY_FIELDS}
+        )
+    return parsed_summaries
+
+
+def get_data(database: str, search_term: str) -> pd.DataFrame:
+    """Retrieve data from the Entrez database based on a search term.
+
+    Parameters
+    ----------
+    database : str
+        Name of the Entrez database.
+    search_term : str
+        The term used to search the specified database.
+
+    Returns
+    -------
+    pd.DataFrame
+        A pandas DataFrame containing the retrieved data.
+    """
+    uids = _search(database, search_term)
+    print(f"Total number of documents returned for the above search query: {len(uids)}")
```

Comment:
Maybe use `logging` instead of print for this.

Reply:
Yes, makes sense. I will replace it with the `logging` module.

Reply:
Hi @lsetiawan, I've made the above changes. Could you please check them? Thank you!